pkg/cable/libreswan/libreswan.go

```diff
@@ -173,15 +214,42 @@ func (i *libreswan) GetName() string {
 
 // Init initializes the driver with any state it needs.
 func (i *libreswan) Init() error {
-	// Write the secrets file:
-	// %any %any : PSK "secret"
-	file, err := os.Create(RootDir + "/etc/ipsec.d/submariner.secrets")
-	if err != nil {
-		return errors.Wrap(err, "error creating the secrets file")
-	}
-	defer file.Close()
+	logger.Infof("Initializing libreswan driver with authentication mode: %s", i.authMode)
+
+	if i.authMode == AuthModePSK {
+		logger.Info("Setting up PSK authentication")
+
+		// Write the secrets file: %any %any : PSK "secret"
+		file, err := os.Create(RootDir + "/etc/ipsec.d/submariner.secrets")
+		if err != nil {
+			return errors.Wrap(err, "error creating the secrets file")
+		}
+		defer file.Close()
+
+		fmt.Fprintf(file, "%%any %%any : PSK \"%s\"\n", i.secretKey)
+	} else if i.authMode == AuthModeCert {
+		logger.Info("Setting up certificate authentication")
+
+		logger.Infof("Starting certificate creation with Private IPs %s Public IPs %s", i.localEndpoint.PrivateIPs, i.localEndpoint.PublicIPs)
+		sanIPs := make([]string, 0, len(i.localEndpoint.PrivateIPs)+len(i.localEndpoint.PublicIPs))
+		sanIPs = append(sanIPs, i.localEndpoint.PrivateIPs...)
+		sanIPs = append(sanIPs, i.localEndpoint.PublicIPs...)
+
+		i.stopCh = make(chan struct{})
 
-	fmt.Fprintf(file, "%%any %%any : PSK \"%s\"\n", i.secretKey)
+		var err error
+
+		i.certificateController = NewCertificateHandler(i.localEndpoint.ClusterID)
+
+		err = i.signingRequestor.Issue(context.TODO(), "submariner-certificate", sanIPs, i.certificateController.OnSignedCallback)
+		if err != nil {
+			logger.Warningf("Unable to issue certificate: %v", err)
+		}
+
+		logger.Info("Started certificate syncers and controller")
+	} else {
+		return fmt.Errorf("unsupported authentication mode: %s", i.authMode)
+	}
```

Comment:
We've already validated the mode at this point.

```suggestion
	}
```